### PR TITLE
perf: enable optimizations for Check

### PR DESCRIPTION
**Pull request by jpadilla**
## Description

This PR introduces a new experimental flag `enable-check-optimizations` to opt in to performance optimizations previously introduced between v1.5.6 and v1.5.8 and later removed in v1.5.9.

## References

https://github.com/openfga/openfga/pull/1837
https://github.com/openfga/openfga/pull/1841

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by adriantam — APPROVED** [review]

File: internal/graph/check.go [hunk 1]
```diff
@@ -1156,7 +1154,15 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
...
 			defer filteredIter.Stop()
 
-			return c.checkUsersetSlowPath(ctx, req, filteredIter)
+			resolver := c.checkUsersetSlowPath
+
+			if c.optimizationsEnabled {
+				if typesys.UsersetCanFastPath(directlyRelatedUsersetTypes, tuple.GetType(reqTupleKey.GetUser())) {
+					resolver = c.checkUsersetFastPath
+				}
+			}
```
Comment (sergiught) [inline internal/graph/check.go:1163]: We can reduce an indentation level here if we use &&.
```suggestion
	if c.optimizationsEnabled &&
		typesys.UsersetCanFastPath(
			directlyRelatedUsersetTypes,
			tuple.GetType(reqTupleKey.GetUser()),
		) {
		resolver = c.checkUsersetFastPath
	}
```

**Review by sergiught — APPROVED** [review]
Only adding some suggestions, otherwise LGTM 👍🏻

File: internal/graph/check.go (hunk 1 above)
Comment (jpadilla) [reply internal/graph/check.go:1161]: thanks for the suggestion, i think the current way is a bit more clear.

File: internal/graph/check.go [hunk 2]
```diff
@@ -1342,7 +1348,21 @@ func (c *LocalChecker) checkTTU(parentctx context.Context, req *ResolveCheckRequ
 		)
 		defer filteredIter.Stop()
 
-		return c.checkTTUSlowPath(ctx, req, rewrite, filteredIter)
+		resolver := c.checkTTUSlowPath
+
+		if c.optimizationsEnabled {
```
[the remark this answers is not in the document]
Comment (jpadilla) [reply internal/graph/check.go:1351]: thanks for the suggestion, i think the current way is a bit more clear.

**Review by sergiught — APPROVED** [review]

**Pull request merged**